numba/cuda/tests/cudapy/test_math.py

```diff
@@ -515,6 +532,13 @@ def test_math_log1p(self):
         self.unary_template_int64(math_log1p, np.log1p)
         self.unary_template_uint64(math_log1p, np.log1p)
 
+    #---------------------------------------------------------------------------
+    # test_math_remainder
+
+    def test_math_remainder(self):
+        self.binary_template_float32(math_remainder, np.remainder, start=1e-11)
+        self.binary_template_float64(math_remainder, np.remainder, start=1e-11)
```

Comment:
You should also test this with integer arguments. Note that a `start` of `1` is required to avoid doing `0/0` in the test. Because of the way the template is set up, starting with `0` will compare the result of `np.remainder(0, 0)` with integer arguments (which results in `0` and a warning) with the result of the CUDA `math.remainder` implementation - so it may be more convenient to also check the result of `0/0` is a NaN in this test, rather than trying to do a refactor of the test infrastructure for this case:

```suggestion
        self.binary_template_float64(math_remainder, np.remainder, start=1e-11)
        self.binary_template_int64(math_remainder, np.remainder, start=1)
        self.binary_template_uint64(math_remainder, np.remainder, start=1)

    def test_math_remainder_0_0(self):
        @cuda.jit(void(float64[::1], int64, int64))
        def test_0_0(r, x, y):
            r[0] = math.remainder(x, y)

        r = np.zeros(1, np.float64)
        test_0_0[1, 1](r, 0, 0)
        self.assertTrue(np.isnan(r[0]))
```